x-pack/plugins/siem/server/lib/detection_engine/signals/filter_events_with_list.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { get } from 'lodash/fp';
+import { Logger } from 'src/core/server';
+
+import { ListItemArraySchema } from '../../../../../lists/common/schemas/response';
+import { type } from '../../../../../lists/common/schemas/common';
+import { ListClient } from '../../../../../lists/server';
+import { SignalSearchResponse, SearchTypes } from './types';
+import { RuleAlertParams } from '../types';
+
+interface FilterEventsAgainstList {
+  listClient: ListClient;
+  exceptionsList: RuleAlertParams['exceptions_list'];
+  logger: Logger;
+  eventSearchResult: SignalSearchResponse;
+}
+
+export const filterEventsAgainstList = async ({
+  listClient,
+  exceptionsList,
+  logger,
+  eventSearchResult,
+}: FilterEventsAgainstList): Promise<SignalSearchResponse> => {
+  try {
+    if (exceptionsList == null || exceptionsList.length === 0) {
+      return eventSearchResult;
+    }
+
+    // narrow unioned type to be single
+    const isStringableType = (val: SearchTypes) =>
+      ['string', 'number', 'boolean'].includes(typeof val);
+    // grab the signals with values found in the given exception lists.
+    const filteredHitsPromises = exceptionsList.map(async (exceptionItem) => {
+      if (exceptionItem.values == null || exceptionItem.values.length === 0) {
+        throw new Error('Malformed exception list provided');
+      }
+      if (!type.is(exceptionItem.values[0].name)) {
+        throw new Error(`Unsupported list type used, please use one of ${type.keys}`);
```

Comment:
Here you want something like this:

```ts
        throw new Error(
          `Unsupported list type used, please use one of ${Object.keys(type.keys).join()}`
        );
```

Tested this by using stronger `toEqual()` in the test with: 

```ts
      expect(exc.message).toEqual(
        'Failed to query lists index. Reason: Unsupported list type used, please use one of ip,keyword'
      );
```

Otherwise you just get a print out of the error being `Unsupported list type used, please use one of [object Object]` and you don't want that.